samples/sensor/sgp40_sht4x/src/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <stdio.h>
+#include <usb/usb_device.h>
+
+/*
+ * User configurable options
+ */
+#define MEASURE_WITH_COMPENSATION 0
+#define USE_HEATER 0
+
+#if MEASURE_WITH_COMPENSATION
+/* public API for handling T/RH compensated measurements */
+#include <drivers/sensor/sgp40.h>
+#endif
+
+#if USE_HEATER
+/* public API handling the heater of SHT4X devices */
+#include <drivers/sensor/sht4x.h>
+/* Threshold above which the heater will be activated */
+#define HEATER_HUM_THRESH 65
+#endif
+
+#if !DT_HAS_COMPAT_STATUS_OKAY(sensirion_sgp40)
+#error "No sensirion,sgp40 compatible node found in the device tree"
+#endif
+
+#if !DT_HAS_COMPAT_STATUS_OKAY(sensirion_sht4x)
+#error "No sensirion,sht4x compatible node found in the device tree"
+#endif
+
+void check_device(const struct device *dev)
+{
+	if (!device_is_ready(dev)) {
+		printf("Device %s is not ready.\n", dev->name);
+		return;
+	}
+}
+
+
+void main(void)
+{
+
+#if MEASURE_WITH_COMPENSATION
+	struct sgp40_compensation_data comp_data;
+#endif
+	struct sensor_value temp, hum, gas;
+
+	const struct device *sht = DEVICE_DT_GET_ANY(sensirion_sht4x);
+	const struct device *sgp = DEVICE_DT_GET_ANY(sensirion_sgp40);
+
+	check_device(sht);
+	check_device(sgp);
```

Comment:
these functions will allow program to continue even if device is not ready.

Reply:
updated the sample.